### ISSUE TITLE
Data validation strategy

### ISSUE THREAD

**Issue by bokolob** [automerge/automerge#807]
Hello. 
What options or strategies do we have to validate incoming changes?
It’s easy to check user input, but  what can I do with remote? 
The worst thing is that local and remote changes could be valid themselves, but harmful together.

**Comment by davidbrochart** [comment]
In [pycrdt](https://github.com/jupyter-server/pycrdt) I'm experimenting with [pydantic](https://docs.pydantic.dev) models to validate incoming changes, by first applying the changes to a "twin" document of the "real" document, and if the "twin" document is still valid then the changes are applied to the "real" document.
But that is potentially expensive, so I'm interested in other strategies.

**Comment by alexjg** [comment]
Yeah that approach is what I would recommend. We are looking to introduce a "branch" feature which will make this much cheaper and easier.
